pkg/blockdevice/memory_mapped_block_device_unix.go

```diff
@@ -91,3 +91,26 @@ func (bd *memoryMappedBlockDevice) WriteAt(p []byte, off int64) (int, error) {
 func (bd *memoryMappedBlockDevice) Sync() error {
 	return unix.Fsync(bd.fd)
 }
+
+func (bd *memoryMappedBlockDevice) Close() error {
+	var errors []error
+
+	if bd.data != nil {
+		if err := unix.Munmap(bd.data); err != nil {
+			errors = append(errors, util.StatusWrap(err, "Failed to unmap memory region"))
+		}
+		bd.data = nil
+	}
+
+	if bd.fd != -1 {
+		if err := unix.Close(bd.fd); err != nil {
+			errors = append(errors, util.StatusWrap(err, "Failed to close file descriptor"))
+		}
+		bd.fd = -1
+	}
```

Comment:
I'd say, we shouldn't implement a function like this in a restartable manner. Even if `Munmap()` and `Close()` fail, we should assume that the underlying resource has been released and nothing can be done to recover. It's not as if the caller is going to write a for-loop around this to retry. Could you remove the if-statements and field assignments, please?